clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapServerStore.java

```diff
@@ -226,6 +144,37 @@ private void writeUnlockAll() {
     }
   }
 
+  private void handleOversizeMappingException(long key, OversizeMappingException e, LongConsumer operation) {
+    handleOversizeMappingException(key, e, k -> {
+      operation.accept(k);
+      return null;
+    });
+  }
+
+  private <R> R handleOversizeMappingException(long key, OversizeMappingException e, LongFunction<R> operation) {
+    if (handleOversizeMappingException(key)) {
```

Comment:
I'm all in for this refactoring. However, I don't get what it's doing. So maybe commenting on the implementation will be useful. We try to evict until it fits if I got it right. But why are we still throwing an exception after looping to evict? And not trying to reapply the operation. Is this part still broken but at least in a single place?

Reply:
I've added some clarifying javadoc.  The current code is I believe correct. One of the removed duplicates had a bug.